Assets/Minigames/Fight/Scripts/Behavior/Movement/BasicAIMovement.cs

```diff
@@ -33,8 +39,15 @@ public class BasicAIMovement : MonoBehaviour, IPathFinder
     // Destination has been reached if agent is within stopping distance of the final waypoint
     public bool reachedDestination => Vector2.Distance(transform.position, path.vectorPath[path.vectorPath.Count-1]) < stopDistance;
 
-    // Path is invalid if the final waypoint is not within stopping distance of target
-    public bool pathInvalid => path != null && Vector2.Distance(path.vectorPath[path.vectorPath.Count - 1], target) > stopDistance;
+    public bool pathInvalid
+    {
+        get
+        {
+            GraphNode node1 = AstarPath.active.GetNearest(transform.position, NNConstraint.Default).node;
+            GraphNode node2 = AstarPath.active.GetNearest(target, NNConstraint.Default).node;
+            return !PathUtilities.IsPathPossible(node1, node2);
```

Comment:
the "double negative" could get confusing. Could rename to isPathValid and get rid of the "!" here. Unless this is only used to check invalidity